faces-tck/pom.xml

```
@@ -0,0 +1,162 @@
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>unpack</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>fish.payara.jakarta.tests.tck</groupId>
+                                    <artifactId>jakarta-faces-tck</artifactId>
+                                    <type>zip</type>
+                                    <version>${project.version}</version>
+                                    <overWrite>false</overWrite>
+                                    <outputDirectory>${tck.root}</outputDirectory>
+                                </artifactItem>
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>replace-glassfish-hardcoded-exec</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <target>
+                                <echo>Replacing Glassfish hardcoded control with Payara</echo>
+                                <copy file="pom.xml-old-tck-run"
+                                    tofile="${project.build.directory}/faces-tck-${faces.version}/tck/old-tck/run/pom.xml" 
+                                    overwrite="true" />
+                            </target>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>exec-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <executable>${maven.executable}</executable>
+                    <commandlineArgs>${maven.exec.cmdline}</commandlineArgs>
+                    <environmentVariables>
+                        <LC_ALL>C</LC_ALL>
+                    </environmentVariables>
+                </configuration>
```

From a encapsulation perspective it makes more sense to have the execution specific config in the actual execution config, rather than as generic plugin config.
```suggestion
                        <configuration>
                            <executable>${maven.executable}</executable>
                            <commandlineArgs>${maven.exec.cmdline}</commandlineArgs>
                            <environmentVariables>
                                <LC_ALL>C</LC_ALL>
                            </environmentVariables>
                        </configuration>
                    </execution>
                </executions>
```

---

Let's refactor and polish it AFTER it works.